trestle/core/control_reader.py

```diff
@@ -672,6 +672,8 @@ def read_all_implementation_prose_and_header(
         if rules:
             yaml_header[const.COMP_DEF_RULES_TAG] = list(rules.values())
         if params:
+            if not set(rules.keys()).issuperset(params.keys()):
```

Comment:
rules can be None

Reply:
rules can be an empty dict but not None, so this is ok.  Same for params.